### ISSUE TITLE
Liste des petits trucs qu'il reste à faire

### ISSUE THREAD

**Issue by tmielcza** [itheof/42sh-posix#125]
Essayons de recenser les petites choses qui manquent:

- [ ] Si isatty() fail, retourner une erreur (?)
- [ ] Commenter (notamment la boucle d'initialisation des termcap)
- [ ] Virer des commentaires
- [ ] Écrire plein de tests (détailler)
- [ ] Gestion correcte de l'historique (lecture depuis un fichier, ecriture dans le fichier a la sortie)
- [ ] Changer history_add pour decouper les commandes multilignes
- [ ] Si ^D faire un saut de ligne
- [ ] La gestion de l'historique dans l'UI ne marche pas comme attendu pour un faible nombre de commandes

**Comment by Otaekek** [comment]
"La gestion de l'historique dans l'UI ne marche pas comme attendu pour un faible nombre de commandes"
What ?

**Comment by dkrmr** [comment]
La gestion de `echo $?` est demandé par le bareme.

**Comment by tmielcza** [comment]
Mec, c'est toi qui l'as, le barème.

**Comment by dkrmr** [comment]
Oui. Et mon message précédent était une affirmation pas une question.

**Comment by tmielcza** [comment]
Mais c'est vrai. Je pense que le point d'interrogation dans la ligne m'a perturbe.

**Comment by tmielcza** [comment]
Je rajoute qu'il faut supprimer tout plein de règles du lexer (sob) ...
